freqtrade/exchange/exchange.py

```diff
@@ -190,6 +210,7 @@ def _init_ccxt(self, exchange_config: Dict[str, Any], ccxt_module: CcxtModuleTyp
             'secret': exchange_config.get('secret'),
             'password': exchange_config.get('password'),
             'uid': exchange_config.get('uid', ''),
+            'options': exchange_config.get('options', {})
```

Comment:
It's related to this message, which I think that I was misinterpreting now 
https://discord.com/channels/700048804539400213/702584313354518659/866751499204165672

Reply:
This was my assumption / suspicion - and i think it's not a user friendly way to approach this.

The user must already configure multiple other settings (for example `trading_mode`) - and based on this, we should automatically set the correct parameters (futures / margin).
In the end, it's a mapping on what needs to be set (for binance, we need to set this, for other exchanges we don't AFAIK).

So while the "options" is a way to get this done for testing (we can also use `ccxt_config` for that) - the goal should be to not have the user care about this - as this is internal logic we can set based on other settings.

Reply:
Ok, can you be more details about how to use ccxt_config, I don't know how to use it

Reply:
well you can add "any" parameter you need there.

so (for developers, until done properly in a user friendly way) - you can do something like the following:

``` jsonc
"ccxt_config": {
  "rateLimit": //...,
  "options": {
     //<whatever options>
  }
}
```

ccxt_config will be used as kwargs to ccxt  (both sync and async instances). There's also seperate configurations for the 2 instances, but you'll not need that for the moment.

Reply:
Couldn't really think of a way to do this with a computed property, because it changes depending on the trading mode, but I think a computed property is better anyway, because ccxt_config is only ever accessed when it's copied

Reply:
what you did looks to work :+1: 

what i'm asking myself though - should binance be using the `binancecoinm` class for futures?

the class itself (https://github.com/ccxt/ccxt/blob/master/js/binancecoinm.js) doesn't look to add too much (actually nothing we'll be using, at least not for the moment) - but it's just something to keep in mind / consider.

Reply:
> what you did looks to work 👍
> 
> what i'm asking myself though - should binance be using the `binancecoinm` class for futures?
> 
> the class itself (https://github.com/ccxt/ccxt/blob/master/js/binancecoinm.js) doesn't look to add too much (actually nothing we'll be using, at least not for the moment) - but it's just something to keep in mind / consider.

USD-M futures would be the one we want I believe, but I'm thinking that everything in here should work for COIN-M as well, if we hard code it in then that might limit the ability to use the other (COIN-M in this case)